parser/src/test/java/com/microsoft/gctoolkit/parser/ConcurrentMarkSweepPhaseParserRulesTest.java

```diff
@@ -17,34 +19,37 @@ public class ConcurrentMarkSweepPhaseParserRulesTest implements CMSPatterns {
 
     @Test
     public void testCMSParseRules() {
-        for (int i = 0; i < rules.length; i++)
+        assertEquals(rules.length,lines.length, "rules length != # of groups of lines to be test");
+        for (int i = 0; i < rules.length; i++) {
             for (int j = 0; j < lines.length; j++) {
-                GCLogTrace trace = rules[i].parse(lines[j]);
-                if (trace != null) {
-                    assertEquals(i, j, i + " captured " + j);
+                int captured = captureTest(rules[i], lines[j]);
+                if (i == j) {
+                    assertEquals(captured, lines[j].length, i + " failed to captured it's lines");
                 } else {
-                    assertTrue(i != j, i + " captured " + j);
+                    assertEquals(0, captured, i + " captured " + j);
                 }
             }
+        }
 
         assertTrue(true);
     }
 
 
     /* Code that is useful when testing individual records */
 
-    @Test
+    //@Test
```

Comment:
Can you ATIgnore but still run from your IDE?

Reply:
ATIgnore is not being recognized in my IDE...

Reply:
@Ignore I mean..

Reply:
@kcpeppe you want `@Test @Ignore`